src/se/vidstige/jadb/HostConnectToRemoteTcpDevice.java

```diff
@@ -0,0 +1,76 @@
+package se.vidstige.jadb;
+
+import se.vidstige.jadb.entities.TcpAddressEntity;
+
+import java.io.IOException;
+
+class HostConnectToRemoteTcpDevice {
+    private final Transport transport;
+    private final ResponseValidator responseValidator;
+
+    HostConnectToRemoteTcpDevice(Transport transport) {
+        this.transport = transport;
+        this.responseValidator = new ResponseValidatorImp();
+    }
+
+    //Visible for testing
+    HostConnectToRemoteTcpDevice(Transport transport, ResponseValidator responseValidator) {
+        this.transport = transport;
+        this.responseValidator = responseValidator;
+    }
+
+    TcpAddressEntity connect(TcpAddressEntity tcpAddressEntity)
+            throws IOException, JadbException, ConnectionToRemoteDeviceException {
+        transport.send(String.format("host:connect:%s:%d", tcpAddressEntity.getHost(), tcpAddressEntity.getPort()));
```

Comment:
Well, it's only 68 lines of code which is not too crazy if you ask me. Yes, the unit tests are very nice, albeit they do not test very much. I like the tests in `MockedTestCases` (This class, however,  desperately needs a split) as I set up exactly what a real adb server would respond with. E.g. For this case we could test "What happens when I send 'host:connect:abc:123' to adb? How does jadb behave when it responds with an error", etc.

Checkout  the [`testPushToInvalidPath`](https://github.com/vidstige/jadb/blob/master/test/se/vidstige/jadb/test/unit/MockedTestCases.java#L79) test for example